File: Patrones/Inmutable/C++/src/Circle.cpp
#include <Circle.h>

Circle::Circle( const double & radius, Color * color, Point * origin){
	this->circleData = new CircleData(radius, color, origin);
}

double Circle::getCircumference(){
	return this->circleData->getRadius() * M_PI;
}

double Circle::getDiameter() {
	return this->circleData->getRadius() * 2;
}
	
string Circle::getColor(){
	return this->circleData->getColor()->toString();
}

Circle::~Circle(){
	delete circleData;
}

File: Patrones/Inmutable/Java/Inmutable/src/example/Circle.java
package example;

public class Circle {
	private CircleData circleData;
	
	public Circle(double radius, Color color, Point origin) {
		this.circleData = new CircleData(radius, color, origin);
	}
	
	public double getCircumference() {
		return this.circleData.getRadius() * Math.PI;
	}
	
	public double getDiameter() {
		return this.circleData.getRadius() * 2;
	}
}